frank-flow/src/frontend/src/app/editor/explorer/add-dialog/add-dialog.component.ts

```diff
@@ -28,8 +28,62 @@ export class AddDialogComponent {
     this.currentDirectory = this.codeService.currentDirectory;
   }
 
+  findDuplicateFileName(fileName: string, directory: any): boolean {
+    if (directory != null && fileName != null) {
+      const foundFile = directory._files.find(
+        (file: string) => file === fileName
+      );
+      if (foundFile) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
You should return the statement. Returning true or false based on an if check is not efficient.

Why are you doing this if it can be shorter and cleaner?

Reply:
fixed